commands/set_command.py

```diff
@@ -0,0 +1,105 @@
+from commands.Icommand import ICommand
+from response.exceptions import *
+import os, json
+from commands.schema_keys import SchemaKeys
+
+parent_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+
+class SetCommand(ICommand):
+    def __init__(self, db, table, value): 
+        self.db = db
+        self.table = table
+        self.value = value
+        self.__validate_path() 
+        self.__validate_value()
+
+    def __validate_path(self):
+        if self.db is None:
+            raise MissingParameterError("Database name is missing.")
+        if self.table is None:
+            raise MissingParameterError("Table name is missing.")
+
+        db_path = os.path.join(parent_dir, self.db)
+        if not os.path.exists(db_path):
+            raise InvalidParameterError("Database does not exist.")
+
+        table_path = os.path.join(db_path, self.table)
+        if not os.path.exists(table_path):
+            raise InvalidParameterError("Table does not exist.")
+
+    def __validate_value(self):
+        if(self.value == None):
+            raise MissingParameterError("Value to be set is missing.")
+
+        self.values = eval(self.value)
+        file = open(os.path.join(parent_dir, self.db, self.table, self.table + "_schema.json"), "r")

```

Comment:
I see you read the schema file here.
Now the logic of read and write schema in scattered across multiple classes.
What I would suggest to have 
```
Class Table
     self.tableMetadata
    
    write(data, query)
     Read(query)
     Read(primary_key)
     Serialize()

 class Metadata:
       self.tablename
       self.primary_key
       self.indices
       
       def serialize(self):
              // write metadata in the file 
               For index in indices:
                     Index.serialize()

class Index
self.name

def serialize 

  
      
```

Reply:
That what I explained last session. This will be more cleaner and help you to use the code across the project.